pkg/driver/centos.go

```diff
@@ -10,8 +11,15 @@ type CentOSDriver struct {
 	systemdUnitOverwriteDriver
 }
 
-func (c CentOSDriver) InstallDocker() error {
-	return executil.ExecPipe("/bin/sh", "-c", "curl -sSL https://get.docker.com/ | sh")
+func (c CentOSDriver) InstallDocker(azureEnv string) error {
+	switch azureEnv {
+		case "AzureCloud":
+			return executil.ExecPipe("/bin/sh", "-c", "curl -sSL https://get.docker.com/ | sh")
+		case "AzureChinaCloud":
+			return executil.ExecPipe("/bin/sh", "-c", "curl -sSL https://mirror.azure.cn/repo/install-docker-engine.sh | sh -s -- --mirror AzureChinaCloud")
```

Comment:
@ahmetalpbalkan (not an endorsement, but I think it's fine if you want to use the version from `master` until that's released)